lib/logstash/inputs/syslog.rb

```diff
@@ -74,30 +77,65 @@ class LogStash::Inputs::Syslog < LogStash::Inputs::Base
   #
   config :locale, :validate => :string
 
-  public
-  def register
-    @metric_errors = metric.namespace(:errors)
+  def initialize(*params)
+    super
+
+    @priority_key = ecs_select[disabled:'priority', v1:'[log][syslog][priority]']
+    @facility_key = ecs_select[disabled:'facility', v1:'[log][syslog][facility][code]']
+    @severity_key = ecs_select[disabled:'severity', v1:'[log][syslog][severity][code]']
+
+    @facility_label_key = ecs_select[disabled:'facility_label', v1:'[log][syslog][facility][name]']
+    @severity_label_key = ecs_select[disabled:'severity_label', v1:'[log][syslog][severity][name]']
+
+    @source_key = ecs_select[disabled:'host', v1:'[source][ip]']
+
+    @grok_pattern ||= ecs_select[
+        disabled:"<%{POSINT:#{@priority_key}}>%{SYSLOGLINE}",
+        v1:"<%{POSINT:#{@priority_key}:int}>%{SYSLOGLINE}"
+    ]
 
     @grok_filter = LogStash::Filters::Grok.new(
-      "overwrite" => @syslog_field,
-      "match" => { @syslog_field => @grok_pattern },
-      "tag_on_failure" => ["_grokparsefailure_sysloginput"],
+        "overwrite" => @syslog_field,
+        "match" => { @syslog_field => @grok_pattern },
+        "tag_on_failure" => ["_grokparsefailure_sysloginput"],
+        "ecs_compatibility" => ecs_compatibility # use ecs-compliant patterns
     )
 
+    @grok_filter_exec = ecs_select[
+        disabled: -> (event) { @grok_filter.filter(event) },
+        v1: -> (event) {
+          event.set('[event][original]', event.get(@syslog_field))
+          @grok_filter.filter(event)
+        }
+    ]
+
     @date_filter = LogStash::Filters::Date.new(
```

Comment:
NOTE: the `ecs_compatibility` flag isn't being passed to the date filter.
Should be fine as ECS support is simply a no-op for the date filter.